pkg/networkmanager/wireguard.jsx

```diff
@@ -0,0 +1,285 @@
+import { Grid } from '@patternfly/react-core/dist/esm/layouts/Grid/index.js';
+import { HelperText, HelperTextItem } from '@patternfly/react-core/dist/esm/components/HelperText/index';
+import { InputGroup } from '@patternfly/react-core/dist/esm/components/InputGroup/index.js';
+import { MinusIcon } from '@patternfly/react-icons';
+import { TextInput } from '@patternfly/react-core/dist/esm/components/TextInput/index.js';
+
+import { Name, NetworkModal, dialogSave } from "./dialogs-common";
+import { ModelContext } from './model-context';
+import { useDialogs } from 'dialogs.jsx';
+
+import './wireguard.scss';
+
+const _ = cockpit.gettext;
+
+function addressesToString(addresses) {
+    return addresses.map(address => address[0] + "/" + address[1]).join(", ");
+}
+
+function stringToAddresses(str) {
+    return str.split(/[\s,]+/).map(strAddress => {
+        const parts = strAddress.split("/");
+        if (parts.length > 2) {
+            throw new Error(_("Addresses is not formatted correctly"));
+        }
+        const [ip, prefix] = parts;
+        const defaultPrefix = "32";
+        // Gateway usually conflicts with routing that NetworkManager configures for WireGuard interfaces
+        // So it should not be set in that case or more accurately set to 0 i.e. "0.0.0.0" in string format
+        const gateway = "0.0.0.0";
+        return [ip, prefix ?? defaultPrefix, gateway];
+    });
+}
+
+export function WireGuardDialog({ settings, connection, dev }) {
+    const Dialogs = useDialogs();
+    const idPrefix = "network-wireguard-settings";
+    const model = useContext(ModelContext);
+
+    const [iface, setIface] = useState(settings.connection.interface_name);
+    const [privateKey, setPrivateKey] = useState("");
+    const [publicKey, setPublicKey] = useState("");
+    const [listenPort, setListenPort] = useState(settings.wireguard.listen_port);
+    const [addresses, setAddresses] = useState(addressesToString(settings.ipv4.addresses));
+    const [dialogError, setDialogError] = useState("");
+    const [peers, setPeers] = useState(settings.wireguard.peers.map(peer => ({ ...peer, allowedIps: peer.allowedIps.join(",") })));
+
+    // Additional check for `wg` after install_dialog for non-packagekit and el8 environments
+    useEffect(() => {
+        async function checkWireguardPackage() {
+            try {
+                await cockpit.script("command -v wg");
+            } catch (e) {
+                setDialogError(_("wireguard-tools package is not installed"));
+            }
+        }
+
+        checkWireguardPackage();
+    }, []);
+
+    // HACK: Fetch private-key in React, not in DBus signal handler to avoid infinite loop
+    // See https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/issues/1336
+    useEffect(() => {
+        async function getPrivateKey() {
+            const objpath = connection[" priv"].path;
```

Comment:
Hmm, that extra space before " priv" *so* looks like a typo. I suppose it's not (otherwise the dialog test would fail), but out of interest -- where does this come from?

Reply:
Right, and everything else uses `priv(obj)`, could this bit use that, too?

Reply:
Ack, thanks!